### PR TITLE
feat: [CDS-39800]: Config files enabled for service V2 creation and readonly mode for all the deployment types

### PR DESCRIPTION
**Pull request by bhavya-sinha**
### Summary
Config files enabled for service V2 creation and readonly mode for all the deployment types

#### Screenshots
https://user-images.githubusercontent.com/75243226/182124278-51aa90b4-9131-43c3-955a-f3505380c498.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [x] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by bhavya-sinha** [comment]
retrigger jest

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/KubernetesServiceSpecInputSetMode.tsx [hunk 1]
```diff
@@ -40,7 +40,7 @@ export interface KubernetesInputSetProps {
   formik?: any
   allowableTypes: AllowedTypes
 }
-const KubernetesServiceSpecInputSetModeFormikForm = (props: KubernetesInputSetProps): React.ReactElement => {
+const GenericServiceSpecInputSetModeFormikForm = (props: KubernetesInputSetProps): React.ReactElement => {
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/KubernetesServiceSpecInputSetMode.tsx:43]: This has already been renamed to Generic. So, we might as well move this to Common folder.

File: src/modules/75-cd/components/PipelineSteps/index.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ import { HttpStep } from './HttpStep/HttpStep'
 import { K8RolloutDeployStep } from './K8sRolloutDeployStep/K8sRolloutDeployStep'
 import { ShellScriptStep } from './ShellScriptStep/ShellScriptStep'
 import { KubernetesInfraSpec } from './KubernetesInfraSpec/KubernetesInfraSpec'
-import { KubernetesServiceSpec } from './K8sServiceSpec/K8sServiceSpec'
+import { GenericServiceSpec } from './K8sServiceSpec/K8sServiceSpec'
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/index.ts:16]: This can be kept as `K8sServiceSpec`

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by litvinovstas — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/index.ts (hunk 2 above)
Comment (bhavya-sinha) [reply src/modules/75-cd/components/PipelineSteps/index.ts:16]: This is used in nativehelm as well

**Pull request merged**